internal/transport/http2_client.go

```diff
@@ -991,7 +1003,10 @@ func (t *http2Client) Close(err error) {
 		t.kpDormancyCond.Signal()
 	}
 	t.mu.Unlock()
-	t.controlBuf.finish()
+	t.controlBuf.put(&goAway{code: http2.ErrCodeNo, debugData: []byte("graceful_shutdown"), closeConnErr: errors.New("graceful_shutdown")})
```

Comment:
It would be nice if we could send more info in the goAway about the graceful shutdown


```suggestion
	t.controlBuf.put(&goAway{code: http2.ErrCodeNo, debugData: []byte(fmt.Sprintf("client shutdown with: %v", err)), closeConnErr: err})
```